src/components/UserActivationPage/index.jsx

```diff
@@ -16,50 +17,39 @@ const USER_ACCOUNT_POLLING_TIMEOUT = 5000;
 
 const UserActivationPage = ({
   match,
-  authentication,
-  userAccount,
-  fetchUserAccount,
 }) => {
+  const user = getAuthenticatedUser();
+  const {
+    username, roles, isActive,
+  } = user;
   const [isPollingUserAccount, setIsPollingUserAccount] = useState(false);
   const enterpriseSlug = useMemo(
     () => match.params.enterpriseSlug,
     [match.params],
   );
   const { addToast } = useContext(ToastsContext);
 
-  useEffect(
-    () => {
-      let timeout;
-      if (authentication?.username && userAccount?.loaded && !userAccount?.isActive) {
-        // user is authenticated and we finished hydrating the full user metadata, but
-        // the user has not verified their email address. we start polling for their user
-        // metadata every USER_ACCOUNT_POLLING_TIMEOUT milliseconds.
-        setIsPollingUserAccount(true);
-        timeout = setTimeout(() => {
-          fetchUserAccount(authentication.username);
-        }, USER_ACCOUNT_POLLING_TIMEOUT);
-      }
-      return () => {
-        if (timeout) {
-          clearInterval(timeout);
-        }
-      };
-    },
-    [authentication, userAccount],
-  );
+  useInterval(() => {
+    if (username && !isActive) {
```

Comment:
Why do hydration criteria differ from component to component? The previous one uses email for this?

Reply:
That component cares if the email is there, this one cares about whether the account is active or not. Both are provided by hydration. (Not sure if this explains it...)